airbyte-integrations/base-python/airbyte_protocol/entrypoint.py

```diff
@@ -90,12 +90,12 @@ def start(self, args):
                     sys.exit(1)
             elif cmd == "discover":
                 catalog = self.source.discover(logger, config_container)
-                print(catalog.serialize())
+                print(catalog.json(exclude_unset=True))
                 sys.exit(0)
             elif cmd == "read":
                 generator = self.source.read(logger, config_container, parsed_args.catalog, parsed_args.state)
                 for message in generator:
-                    print(message.serialize())
+                    print(message.json(exclude_unset=True))
```

Comment:
there's no way to set this as a default is there so we don't need to include it in every invocation.

Reply:
Not really. We could override it for individual classes but that feels super hacky. It also isn't strictly _wrong_ if it outputs with all of the null fields. It shouldn't break anything.